nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/json/JsonTreeReader.java

```diff
@@ -59,32 +61,62 @@
         + "If an array is encountered, each element in that array will be treated as a separate record. "
         + "If the schema that is configured contains a field that is not present in the JSON, a null value will be used. If the JSON contains "
         + "a field that is not present in the schema, that field will be skipped. "
-    + "See the Usage of the Controller Service for more information and examples.")
+        + "See the Usage of the Controller Service for more information and examples.")
 @SeeAlso(JsonPathReader.class)
 public class JsonTreeReader extends SchemaRegistryService implements RecordReaderFactory {
 
     private volatile String dateFormat;
     private volatile String timeFormat;
     private volatile String timestampFormat;
+    private volatile String startingFieldName;
+    private volatile StartingFieldStrategy startingFieldStrategy;
+
+    public static final PropertyDescriptor STARTING_FIELD_STRATEGY = new PropertyDescriptor.Builder()
+            .name("starting-field-strategy")
+            .displayName("Starting Field Strategy")
+            .description("Start processing from the root node or from a specified nested node.")
+            .required(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .defaultValue(StartingFieldStrategy.ROOT_NODE.name())
+            .allowableValues(
+                    Arrays.stream(StartingFieldStrategy.values()).map(startingStrategy ->
+                            new AllowableValue(startingStrategy.name(), startingStrategy.name(), startingStrategy.getDescription())
```

Comment:
Recommend adding a `getDisplayName()` and `StartingFieldStrategy` and referencing it when constructing the AllowableValue to make it more readable:
```suggestion
                            new AllowableValue(startingStrategy.name(), startingStrategy.getDisplayName(), startingStrategy.getDescription())
```